test/src/tpm/formatting_tools_test.c

```diff
@@ -2136,3 +2149,59 @@ void test_concat(void)
   CU_ASSERT(concat(&dest, &dest_len, chile, -1) == 1);
   free(dest);
 }
+
+//----------------------------------------------------------------------------
+// test_verifyConvertionStringDigest()
+//----------------------------------------------------------------------------
+void test_verifyConvertionStringDigest(void)
+{
+  TSS2_SYS_CONTEXT *sapi_ctx = NULL;
+
+  init_tpm2_connection(&sapi_ctx);
+  TPML_PCR_SELECTION pcrs_struct = {.count = 0, };
+
+  TPM2B_DIGEST digest;
+  TPM2B_DIGEST test_digest;
+  char *string;
+
+  create_policy_digest(sapi_ctx, pcrs_struct, &digest);
+  CU_ASSERT(digest.size != 0);
+  string = (char *) malloc((digest.size *2) + 1);
+
+  CU_ASSERT(convert_digest_to_string(&digest, string) == 0);
+  CU_ASSERT(string != NULL);
+  CU_ASSERT(convert_string_to_digest(string, &test_digest) == 0);
+  CU_ASSERT(test_digest.size != 0);
+  free(string);
+}
+
+//----------------------------------------------------------------------------
```

Comment:
This is really just an observation, but for some reason, we put the marshal/unmarshal (pack/unpack) functionality in a separate source file (marshalling_tools.c), but, we combined these tests with those for formatting_tools.c in formatting_tools_test.c instead of creating a marshalling_tools_test.c (this includes tests for utilities that are not part of this pull request).

Reply:
The marshaling _tools.c used to be part of the formatting_tools.c. We should update the tests to ad a marshalling_tools_test.c. I have submitted a issue to do this.